ephios/extra/auth.py

```diff
@@ -1,14 +1,36 @@
+from typing import Any, Dict
+from urllib.parse import urljoin
+
+import jwt
+from django.conf import settings
 from django.contrib.auth import get_user_model
-from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from django.contrib.auth.backends import ModelBackend
+from django.core.exceptions import SuspiciousOperation
+from django.urls import reverse
+from jwt import InvalidTokenError
+from oauthlib.oauth2 import WebApplicationClient
+from requests_oauthlib import OAuth2Session
+from urllib3.exceptions import RequestError
+
+from ephios.core.models.users import IdentityProvider
 
 
-class EphiosOIDCAB(OIDCAuthenticationBackend):
+class EphiosOIDCAB(ModelBackend):
+    def decode_jwt_token(self, token: str) -> Dict[str, Any]:
+        jwks_client = jwt.PyJWKClient(self.provider.jwks_uri)
+        header = jwt.get_unverified_header(token)
+        key = jwks_client.get_signing_key(header["kid"]).key
+        decoded = jwt.decode(token, key, [header["alg"]], audience=self.provider.client_id)
+        return decoded
+
     def create_user(self, claims):
         user = get_user_model()(email=claims.get("email"))
```

Comment:
So at the moment ephios can only use email as an identifying attribute. What if people want to use oidc providers where users can change their email freely? This could be used to hijack another ephios account!

Reply:
hm, I think we could just add a hint about it so that people are aware when cofiguring a provider. We could also go to the full length and add another model with some kind of OIDC association, but that opens another set of problems. And I think all usecases that we have seen so far don't require such a complicated setup

Reply:
add security hint